source/CMakeLists.txt

```diff
@@ -49,4 +50,8 @@ endif()
 
 if(BUILD_ROBOT_MODEL)
   add_subdirectory(robot_model)
+endif()
+
+if(BUILD_DEMOS)
```

Comment:
Also, couldn't the demos go to the top level instead of source? They would be more visible there. But it guess this doesn't work because of the CMake...

Reply:
I guess we could as well. In that case it would probably have its own `CMakeLists.txt` independent of the one in source

Reply:
I agree completely, my first comment would have been that demos should live outside of source, and install / import control libraries as a separate project / set of projects, rather than being a submodule. 

This also works better in the sense that the demos can have their own dockerfiles and build systems (e.g. one with CMake, one with catkin, etc), and are not "installed" by default if you just take the zip source archive

Reply:
Good I will move them. I don't think they need their own Dockerfile though. Multi stage build works actually very nicely with existing one (see new commit). But I agree on the architecture.